Return Type for Dialect.create_connect_args is incorrect

### Ensure stubs packages are not installed

- [X] No sqlalchemy stub packages is installed (both `sqlalchemy-stubs` and `sqlalchemy2-stubs` are not compatible with v2)

### Verify if the api is typed

- [X] The api is not in a module listed in [#6810](https://github.com/sqlalchemy/sqlalchemy/issues/6810) so it should pass type checking

### Describe the typing issue

The type signature for `create_connect_args` reports that it returns `ConnectArgsType` (`Tuple[Tuple[str], MutableMapping[str, Any]]`); however, the example docs indicate that tuple in the first element of the return tuple may be empty:
https://github.com/sqlalchemy/sqlalchemy/blob/6d5cdd1f59415ea8f08efe0f40126b8c384d1125/lib/sqlalchemy/engine/interfaces.py#L1231-L1236
and likewise the default implementation:
https://github.com/sqlalchemy/sqlalchemy/blob/6d5cdd1f59415ea8f08efe0f40126b8c384d1125/lib/sqlalchemy/engine/default.py#L617-L621
In addition, both cases use lists rather than tuples

### To Reproduce

```python
from sqlalchemy.engine.default import DefaultDialect
from typing import Tuple, Dict, Any


class MyDialect(DefaultDialect):
    def create_connect_args(self, url) -> Tuple[Tuple[str], Dict[str, Any]]:
        opts = url.translate_connect_args()
        opts.update(url.query)
        return ((), opts)
```


### Error

```
test.py:9: error: Incompatible return value type (got "tuple[tuple[], Any]", expected "tuple[tuple[str], dict[str, Any]]")  [return-value]
``` 

### Versions

- OS: MacOS
- Python: 3.10.2
- SQLAlchemy: 2.0.17
- Type checker (eg: mypy 0.991, pyright 1.1.290, etc): mypy 1.4.1


### Additional context

_No response_

--- a/lib/sqlalchemy/engine/default.py
+++ b/lib/sqlalchemy/engine/default.py
@@ -618,7 +618,7 @@ class DefaultDialect(Dialect):
         # inherits the docstring from interfaces.Dialect.create_connect_args
         opts = url.translate_connect_args()
         opts.update(url.query)
-        return [[], opts]
+        return ((), opts)
 
     def set_engine_execution_options(
         self, engine: Engine, opts: Mapping[str, Any]

--- a/lib/sqlalchemy/engine/interfaces.py
+++ b/lib/sqlalchemy/engine/interfaces.py
@@ -71,7 +71,7 @@ if TYPE_CHECKING:
     from ..sql.type_api import _TypeMemoDict
     from ..sql.type_api import TypeEngine
 
-ConnectArgsType = Tuple[Tuple[str], MutableMapping[str, Any]]
+ConnectArgsType = Tuple[Sequence[str], MutableMapping[str, Any]]
 
 _T = TypeVar("_T", bound="Any")
 
@@ -1233,7 +1233,7 @@ class Dialect(EventTarget):
             def create_connect_args(self, url):
                 opts = url.translate_connect_args()
                 opts.update(url.query)
-                return [[], opts]
+                return ((), opts)
 
         :param url: a :class:`.URL` object
 

